Patch from developer to fix build on Mac OS X < 10.7 which doesn't have strnlen
--- dcraw.c.orig	2018-06-01 16:20:44.000000000 -0500
+++ dcraw.c	2018-06-10 22:54:25.000000000 -0500
@@ -9923,7 +9923,8 @@
   if (type == 1 && count <= 4)
     FORC(4) tt->val.c[c] = val >> (c << 3);
   else if (type == 2) {
-    count = strnlen((char *)th + val, count-1) + 1;
+    ((char *)th)[val+count-1] = 0;
+    count = strlen((char *)th + val) + 1;
     if (count <= 4)
       FORC(4) tt->val.c[c] = ((char *)th)[val+c];
   } else if (type == 3 && count <= 2)
